library/system/io.lean

```diff
@@ -3,17 +3,17 @@ Copyright (c) 2016 Microsoft Corporation. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Luke Nelson, Jared Roesch and Leonardo de Moura
 -/
-constant {u} io : Type u → Type u
-constant io.functor : functor io
-constant io.monad   : monad io
+meta constant {u} io : Type u → Type u
+meta constant io.functor : functor io
```

Comment:
Here are different scenarios for `meta/untrusted`.

1- I want to use unbounded recursion when defining `foo`. In this scenario, I cannot prove anything about `foo`. This is fine if `foo` is a function used for meta-programming (e.g., it is a tactic). 

2- We add a (meta-) primitive `foo` that breaks referential transparency. We do it because it is convenient. We really want to mark `foo` as `meta` because it doesn't even make sense to prove any property about `foo`.

3- I want to use some primitive `foo` available in the OS. I manually axiomatize properties of this primitive. We may even execute the code based on `foo` by using the `ffi` infrastructure (when it becomes stable). `foo` and its properties are declared using `constant/axiom`. Now, suppose we define `bla` using `foo`. We can prove things about `bla` using the axioms above. Sure, if we make a mistake in the axioms, we can prove false. My point is, I don't want `bla` to be tagged as `meta/untrusted`. I want it to be a regular definition, and when I type `print axioms bla`, I get `foo` and all its properties.

I think scenarios 1&2 are very different from 3. The scenarios 1&2 already exist in the stdlib. Scenario 3 is very natural, and @jroesch is planning to pursue this direction.